### PR TITLE
Bugfix ui

### PR DESCRIPTION
**Pull request by jgminer**

### CODE REVIEW

File: client/bindings/pillarEditor.js [hunk 1]
```diff
@@ -17,12 +26,12 @@ define(
...
             update: function(element, valueAccessor, allBindings, viewModel, bindingContext) {
                 var editing = ko.unwrap(valueAccessor());
-                var pillarModel = bindingContext.$parents[2];
-
+                var pillarModel = getPillarModel(bindingContext);
                 if (editing) {
                     pillarModel.tableEditing = true;
                 }
-                var index = pillarModel.checkedNodes.indexOf(bindingContext.$parent);
+                var hasProjectIndex = pillarModel.checkedNodes.indexOf(bindingContext.$parent);
```
Comment (nicmoski) [inline client/bindings/pillarEditor.js]: the variable name `hasProjectIndex` seems to indicate that it is a boolean, but it is actually an integer, no?

Comment (jgminer) [inline client/bindings/pillarEditor.js]: the 'has' is referring to hasProject, but I understand how this is unclear. will be an easy fix

File: client/model/pillarModel.js [hunk 2]
```diff
@@ -220,85 +304,73 @@ define( [
...
-                    if (typeof _assoc.projects[data] !== "undefined"){
+                if (typeof modalToHide !== "undefined") {
+                    $("#"+modalToHide).modal('hide');
+                }
+
+                var doesNotAlreadyExist = function(_assoc, num_remaining) {
+                    if (typeof _assoc.projects[project_name] !== "undefined") {
                         swal("Warning", "Project already exists on " + _assoc.name, 'error');
-                        left--;
+                        num_remaining--;
```
Comment (nicmoski) [inline client/model/pillarModel.js]: is `num_remaining` ever used?

Comment (jgminer) [inline client/model/pillarModel.js]: No it was an idea for a future release.

File: client/bindings/pillarEditor.js [hunk 3]
```diff
@@ -3,7 +3,16 @@ define(
         'knockout',
         'jquery'
     ], 
-    function(ko, $) { 
+    function(ko, $) {
+        var getPillarModel = function(bindingContext) {
+            if (bindingContext.$parents[2].constructor.name === 'pillarModel') {
+                return bindingContext.$parents[2];
```
Comment (jalons) [inline client/bindings/pillarEditor.js]: Side question, but what's special about $parents[2] vs $parents[3]?

Comment (jgminer) [inline client/bindings/pillarEditor.js]: In the HTML, I am calling this function is called from two different knockout contexts, and this is how I handle that.

**Pull request merged**